section.go

```diff
@@ -70,6 +74,44 @@ func (w *writer) String(s string) {
 	w.Write(b)
 }
 
+func (w *writer) encodeRanks(ranks [][]float64) error {
```

Comment:
writer is somewhat generic. Maybe instead you can add a function for `encodeGob` and `decodeGob`?

Reply:
I made encodeRanks a function. I left the name because I like that it hides that we use gob, because the caller doesn't need to know.